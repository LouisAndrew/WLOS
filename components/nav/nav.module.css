.container {
  grid-template-rows: auto 1fr auto;
  @apply grid bg-primary-gray px-5 py-5 duration-200 w-full flex-shrink-0
    absolute -left-full top-0 sm:relative sm:-left-5 sm:-top-10 sm:w-max;
}

.container_show_mobile {
  @apply left-0 sm:-left-10;
}

.link_list {
  @apply block mb-auto pl-0;
}

.link_list li {
  @apply my-2;
}

.settings_btn {
  margin-top: auto;
}

.upper_section {
  @apply flex justify-between items-center mb-3;
}

.upper_section_wrap {
  @apply sm:flex-col sm:justify-center;
}

.upper_section_wrap .expand_btn {
  @apply sm:mt-4;
}

.btn svg {
  @apply h-6 w-6 sm:w-8 sm:h-8;
}

.expand_btn {
  @apply rounded hidden sm:flex flex-center h-7 w-7;
}
.close_btn {
  @apply block sm:hidden;
}

.expand_btn svg {
  @apply h-6 w-6;
}
