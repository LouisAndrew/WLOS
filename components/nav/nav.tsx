import React, { FC, useState } from 'react'
import { RiArrowLeftSLine, RiArrowRightSLine, RiCloseLine } from 'react-icons/ri'

import { NavLink } from './nav-link'
import { Avatar } from './avatar'
import styles from './nav.module.css'

export type Props = {
  /**
   * sets if the menu / nav should be shown on mobile
   */
  showMenu: boolean
  /**
   * function to close / hide menu on mobile
   */
  closeMenu: () => void
}

const Nav: FC<Props> = ({ closeMenu, showMenu }) => {
  const [shouldShrinkMenu, setShouldShrinkMenu] = useState(false)

  return (
    <nav className={`${styles.container} ${showMenu ? styles.container_show_mobile : ''}`}>
      <div
        className={`${styles.upper_section} ${shouldShrinkMenu ? styles.upper_section_wrap : ''}`}
      >
        <button
          className={`nav-button ${styles.btn} ${styles.close_btn}`}
          data-testid="close-btn"
          onClick={closeMenu}
        >
          <RiCloseLine />
        </button>
        <Avatar />
        <button
          className={`nav-button ${styles.btn} ${styles.expand_btn}`}
          onClick={() => setShouldShrinkMenu((prev) => !prev)}
          data-testid="shrink-toggle"
        >
          {shouldShrinkMenu ? (
            <RiArrowRightSLine data-testid="right-arrow" />
          ) : (
            <RiArrowLeftSLine data-testid="left-arrow" />
          )}
        </button>
      </div>
      <ul className={styles.link_list}>
        <li>
          <NavLink shouldShrink={shouldShrinkMenu} type="HOME" />
        </li>
        <li>
          <NavLink shouldShrink={shouldShrinkMenu} type="PLANS" />
        </li>
        <li>
          <NavLink shouldShrink={shouldShrinkMenu} type="LOGS" />
        </li>
      </ul>
      <NavLink shouldShrink={shouldShrinkMenu} type="SETTINGS" className={styles.settings_btn} />
    </nav>
  )
}

export default Nav
