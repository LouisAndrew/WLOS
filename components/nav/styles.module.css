.container {
  height: -webkit-fill-content;
  grid-template-rows: auto 1fr auto;
  @apply grid h-screen bg-primary-gray px-5 py-5 duration-200 w-min;
}

.link_list {
  @apply block mb-auto;
}

.link_list li {
  @apply my-2;
}

.settings_btn {
  margin-top: auto;
}
