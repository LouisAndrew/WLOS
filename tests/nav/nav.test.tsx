import { render, fireEvent, cleanup } from '@testing-library/react'

import { Nav } from '@c/nav'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

beforeEach(() => {
  useRouter.mockImplementationOnce(() => ({
    pathname: '/',
  }))
})

jest.mock('@c/nav/nav-link/nav-link', () => jest.fn(() => <div />))

afterEach(cleanup)

const closeMenu = jest.fn()

const Component = <Nav showMenu={true} closeMenu={closeMenu} />

describe('Navigation component', () => {
  it('matches snapshot', () => {
    expect(render(Component, {})).toMatchSnapshot()
  })

  it('render arrow icon based on `shrink` state', () => {
    const result = render(Component)
    const shrinkBtn = result.getByTestId('shrink-toggle')

    const leftArrowTestId = 'left-arrow'
    const rightArrowTestId = 'right-arrow'

    expect(result.getByTestId(leftArrowTestId)).toBeInTheDocument()
    expect(result.queryByTestId(rightArrowTestId)).not.toBeInTheDocument()

    fireEvent.click(shrinkBtn)

    expect(result.queryByTestId(leftArrowTestId)).not.toBeInTheDocument()
    expect(result.getByTestId(rightArrowTestId)).toBeInTheDocument()
  })

  it('should call closeMenu function when close button is clicked', () => {
    const result = render(Component)
    const closeBtn = result.getByTestId('close-btn')

    fireEvent.click(closeBtn)
    expect(closeMenu).toBeCalled()
  })
})
