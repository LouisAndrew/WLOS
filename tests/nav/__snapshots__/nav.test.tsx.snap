// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="container container_show_mobile"
      >
        <div
          class="upper_section "
        >
          <button
            class="nav-button btn close_btn"
            data-testid="close-btn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636z"
                />
              </g>
            </svg>
          </button>
          <div
            class="container"
          />
          <button
            class="nav-button btn expand_btn"
            data-testid="shrink-toggle"
          >
            <svg
              data-testid="left-arrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M10.828 12l4.95 4.95-1.414 1.414L8 12l6.364-6.364 1.414 1.414z"
                />
              </g>
            </svg>
          </button>
        </div>
        <ul
          class="link_list"
        >
          <li>
            <div />
          </li>
          <li>
            <div />
          </li>
          <li>
            <div />
          </li>
        </ul>
        <div />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="container container_show_mobile"
    >
      <div
        class="upper_section "
      >
        <button
          class="nav-button btn close_btn"
          data-testid="close-btn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636z"
              />
            </g>
          </svg>
        </button>
        <div
          class="container"
        />
        <button
          class="nav-button btn expand_btn"
          data-testid="shrink-toggle"
        >
          <svg
            data-testid="left-arrow"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M10.828 12l4.95 4.95-1.414 1.414L8 12l6.364-6.364 1.414 1.414z"
              />
            </g>
          </svg>
        </button>
      </div>
      <ul
        class="link_list"
      >
        <li>
          <div />
        </li>
        <li>
          <div />
        </li>
        <li>
          <div />
        </li>
      </ul>
      <div />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
